core/src/main/java/hudson/PluginManager.java

```diff
@@ -2152,15 +2154,24 @@ public void addNamedClass(String className, Class c) {
             if (FAST_LOOKUP) {
                 for (PluginWrapper p : activePlugins) {
                     try {
-                        Class<?> c = ClassLoaderReflectionToolkit._findLoadedClass(p.classLoader, name);
+                        Class<?> c;
+                        if (p.classLoader instanceof AntWithFindResourceClassLoader) {
+                            c = ((AntWithFindResourceClassLoader) p.classLoader).findLoadedClass2(name);
```

Comment:
To avoid duplication, maybe revert all these changes at call sites, and encode the desired nonreflective behavior into `ClassLoaderReflectionToolkit`? (For clarity, could introduce an `interface` implemented by `AntWithFindResourceClassLoader` which defines the required methods.)

Reply:
yup was thinking about that, the only reason I didn't do that was I was looking to see if we could retire it completely